includes/Hooks.php

```diff
@@ -96,16 +97,39 @@ public function onSpecialPageBeforeExecute( $special, $subPage ) {
 			return true;
 		}
 
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+		$protectedSpecialPages = $config->get( 'CrawlerProtectedSpecialPages' );
+		$denyFast = $config->get( 'CrawlerProtectedSpecialPages' );
+
```

Comment:
Rather than having multiple checks, please add a line to get a version of `$protectedSpecialPages` which has been `tolowercase`ed and had `Special:` stripped from it.

For example:

```php
$result = array_map(
    fn($p) => ($p = strtolower($p)) && strpos($p, NS_SPECIAL_NAME) === 0
        ? substr($p, 8)
        : $p,
    $protectedSpecialPages
);
```